static_frame/core/util.py

```diff
@@ -3186,5 +3187,34 @@ def iloc_to_insertion_iloc(key: int, size: int) -> int:
         raise IndexError(f'index {key} out of range for length {size} container.')
     return key % size
 
+def all_nested_elements(it: any, seen=None) -> tp.Iterable[any]:
+    pass
 
-
+def getsizeof_recursive(obj: any, *, seen=None) -> int:
+    '''
+    Gives the total size of an iterable of elements
+    see also: https://code.activestate.com/recipes/577504/
+    '''
+    total = 0
+    seen = set() if seen is None else seen
+    if id(obj) in seen:
+        return 0
+    seen.add(id(obj))
+    total += getsizeof(obj)
+    # Check if iterable or a string first for fewer isinstance calls on common types
+    if hasattr(obj, '__iter__') and not isinstance(obj, str):
+        if (
+            (isinstance(obj, np.ndarray) and obj.dtype == np.object)
```

Comment:
Swap `obj.dtype == np.object` with `obj.dtype is np.object`?

Reply:
Common in SF and fastest is the following. Note that we cannot do `is` comparisons on dtypes as there are many instances of object dtype...

```suggestion
            (obj.__class__ is np.ndarray and obj.dtype.kind == DTYPE_OBJECT_KIND)
```